src/python/gennylib/auto_tasks.py

```diff
@@ -0,0 +1,443 @@
+    """
+
+    ALL_TASKS = object()
+    VARIANT_TASKS = object()
+    PATCH_TASKS = object()
+
+
+class CLIOperation(NamedTuple):
+    """
+    Represents the "input" to what we're doing"
+    """
+
+    mode: OpName
+    variant: Optional[str]
+    is_legacy: bool
+    output_file_suffix: str
+
+    @property
+    # Kill once we kill is_legacy
+    def repo_root(self) -> str:
+        if self.is_legacy:
+            if self.mode == OpName.VARIANT_TASKS:
+                return "../src/genny/genny"
+            elif self.mode == OpName.ALL_TASKS:
+                return "../src/genny/genny"
+            elif self.mode == OpName.PATCH_TASKS:
+                return "./genny/genny"
+        return "./src/genny"
+
+    @property
+    # Kill once we kill is_legacy
+    def output_file(self) -> str:
+        if self.is_legacy:
+            return os.path.join(self.repo_root, self.output_file_suffix)
+        return self.output_file_suffix
+
+    @staticmethod
+    def parse(argv: List[str], reader: YamlReader) -> "CLIOperation":
+        mode = OpName.ALL_TASKS
+        is_legacy = False
+        variant = None
+        output_file = None
+
+        # Purposefully not using argparse etc - won't need
+        # it once we switch from legacy. Right now only targeting
+        # the exact invocations in system_perf.yml in the mongo repo.
+        if "--generate-all-tasks" in argv:
+            mode = OpName.ALL_TASKS
+            is_legacy = True
+        if "--modified" in argv:
+            mode = OpName.PATCH_TASKS
+            is_legacy = True
+        if "--autorun" in argv:
+            mode = OpName.VARIANT_TASKS
+            is_legacy = True
+        if mode in {OpName.PATCH_TASKS, OpName.VARIANT_TASKS}:
+            variant = argv.index("--variants")
+            variant = argv[variant + 1]
+        if "--output" in argv:
+            output_file = argv[argv.index("--output") + 1]
+        if is_legacy is False:
+            output_file = "./run/build/Tasks/Tasks.json"
+            if argv[1] == "all_tasks":
+                mode = OpName.ALL_TASKS
+            if argv[1] == "patch_tasks":
+                mode = OpName.PATCH_TASKS
+                variant = reader.load("expansions.yml")["build_variant"]
+            if argv[1] == "variant_tasks":
+                mode = OpName.VARIANT_TASKS
+                variant = reader.load("expansions.yml")["build_variant"]
+        return CLIOperation(mode, variant, is_legacy, output_file)
+
+
+class CurrentBuildInfo:
+    def __init__(self, reader: YamlReader):
+        # Just reader.load("expansions.yml") when killing legacy.
+        conts = reader.load_set([f"{b}.yml" for b in {"bootstrap", "runtime", "expansions"}])
+        if "expansions" in conts:
+            conts = conts["expansions"]
+        else:
+            if "bootstrap" not in conts:
+                # Dangerous territory, but this goes away once we're done with legacy.
+                return
```

Comment:
We actually don't have `bootstrap` for the all-tasks case, so it's technically not an error at this point (hard to track down where it would blow up in the future hence the "dangerous territory stuff" 😄). If this were gonna stay I'd probably refactor to not require a `CurrentBuildInfo` for the list-all-tasks case or something.

I added a comment stating that We don't have a "bootstrap" value in legacy list-all-tasks case.